nexus/db-queries/src/db/pool.rs

```diff
@@ -3,108 +3,144 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 //! Database connection pooling
-// This whole thing is a placeholder for prototyping.
-//
-// TODO-robustness TODO-resilience We will want to carefully think about the
-// connection pool that we use and its parameters.  It's not clear from the
-// survey so far whether an existing module is suitable for our purposes.  See
-// the Cueball Internals document for details on the sorts of behaviors we'd
-// like here.  Even if by luck we stick with bb8, we definitely want to think
-// through the various parameters.
-//
-// Notes about bb8's behavior:
-// * When the database is completely offline, and somebody wants a connection,
-//   it still waits for the connection timeout before giving up.  That seems
-//   like not what we want.  (To be clear, this is a failure mode where we know
-//   the database is offline, not one where it's partitioned and we can't tell.)
-// * Although the `build_unchecked()` builder allows the pool to start up with
-//   no connections established (good), it also _seems_ to not establish any
-//   connections even when it could, resulting in a latency bubble for the first
-//   operation after startup.  That's not what we're looking for.
-//
 // TODO-design Need TLS support (the types below hardcode NoTls).
 
 use super::Config as DbConfig;
-use async_bb8_diesel::ConnectionError;
-use async_bb8_diesel::ConnectionManager;
+use crate::db::pool_connection::DieselPgConnector;
+
+use qorb::backend;
+use qorb::policy::Policy;
+use qorb::resolver::{AllBackends, Resolver};
+use qorb::resolvers::dns::{DnsResolver, DnsResolverConfig};
+use qorb::service;
+use std::collections::BTreeMap;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tokio::sync::watch;
 
 pub use super::pool_connection::DbConnection;
 
+type QorbConnection = async_bb8_diesel::Connection<DbConnection>;
+type QorbPool = qorb::pool::Pool<QorbConnection>;
+
 /// Wrapper around a database connection pool.
 ///
 /// Expected to be used as the primary interface to the database.
 pub struct Pool {
-    pool: bb8::Pool<ConnectionManager<DbConnection>>,
+    inner: QorbPool,
 }
 
-impl Pool {
-    pub fn new(log: &slog::Logger, db_config: &DbConfig) -> Self {
-        // Make sure diesel-dtrace's USDT probes are enabled.
-        usdt::register_probes().expect("Failed to register USDT DTrace probes");
-        Self::new_builder(log, db_config, bb8::Builder::new())
-    }
+// Provides an alternative to the DNS resolver for cases where we want to
+// contact the pool directly.
```

Comment:
"contact the pool directly" doesn't make a ton of sense to me in this context --- I'm assuming this is for the case where we use a single DB address, such as in tests?

Reply:
Yeah, this was incorrect phrasing -- it's for cases where we want to contact the **database** directly without performing resolution first, when there's a single DB as you mentioned.

Reply:
(I'll update the comment)